codalab/model/bundle_model.py

```diff
@@ -1047,15 +1048,21 @@ def update_disk_metadata(self, bundle, bundle_location, enforce_disk_quota=False
         Computes the disk use and data hash of the given bundle.
         Updates the database rows for the bundle and user with the new disk use
         """
-        dirs_and_files = None
-        if os.path.isdir(bundle_location):
-            dirs_and_files = path_util.recursive_ls(bundle_location)
+        if bundle_location.startswith("azfs://"):
+            # TODO(Ashwin): add a FileSystems.size() method to Apache Beam to make this less verbose.
+            size = FileSystems.get_filesystem(bundle_location).size(bundle_location)
+            # TODO(Ashwin): use an actual hash instead of the size in the future.
+            data_hash = size
```

Comment:
I don't think we will have a hash in the future since it would require reading everything, so just set it to None?

Reply:
Why not just keep it at `size`? I can remove the TODO comment if that's final, though.